### PR TITLE
Fixes a ... likely important bug with shuttle code....

### PR DESCRIPTION
**Pull request by silicons**
Fuck

### CODE REVIEW

**Comment by kevinz000** [comment]
testing

**Comment by kevinz000** [comment]
aka im still testing this and this might not work but the image is what happens right now i believe.

**Comment by RemieRichards** [comment]
@kevinz000 this was fixed I believe by @ninjanomnom

**Comment by kevinz000** [comment]
@RemieRichards I got runtimes of incorrect turf deletions and after testing it's because the typecache filter list included the turf itself in the list returned, so when turfs are empty()'d they're trying to force-delete themselves without changeTurf.
Also for the other thing, it's still runtiming.

**Comment by kevinz000** [comment]
tested

File: code/modules/shuttle/on_move.dm [hunk 1]
```diff
@@ -13,8 +13,8 @@ All ShuttleMove procs go here
 // Called from the new turf before anything has been moved
 // Only gets called if fromShuttleMove returns true first
 /turf/proc/toShuttleMove(turf/oldT, shuttle_dir)
-	for(var/i in 1 to contents.len)
-		var/atom/movable/thing = contents[i]
+	for(var/i in contents)
+		var/atom/movable/thing = i
```
Comment (ninjanomnom) [inline code/modules/shuttle/on_move.dm]: why was this changed?

Comment (Cyberboss) [inline code/modules/shuttle/on_move.dm]: It's known iterating lists like this is faster

Comment (Cyberboss) [inline code/modules/shuttle/on_move.dm]: But yeah, unrelated change

Comment (ninjanomnom) [inline code/modules/shuttle/on_move.dm]: remie specifically said that the `in 1 to len` loop is faster than the `in list` and had me change a whole bunch of them for that reason
